### PR TITLE
[Kotlin Migration] AndroidUiUtils 

### PR DESCRIPTION
**Pull request by codingtosh**

### CODE REVIEW

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
Doesn't compile

**Comment by codingtosh** [comment]
> Doesn't compile

Hmm, I double checked that everything compiled and built without problems before sending the PR in. 
Going by the logs, 

> /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/MaterialEditTextDialog.kt: (52, 52): Type mismatch: inferred type is EditText? but View was expected

it seems newer commits were made in main after I had committed locally. 

I've update the local codebase and patched this up, should I amend this in the previous commit(changes a different file as well) or make a third commit?

**Comment by david-allison-1** [comment]
> > Doesn't compile
> 
> Hmm, I double checked that everything compiled and built without problems before sending the PR in. Going by the logs,
> 
> > /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/MaterialEditTextDialog.kt: (52, 52): Type mismatch: inferred type is EditText? but View was expected
> 
> it seems newer commits were made in main after I had committed locally.
> 
> I've update the local codebase and patched this up, should I amend this in the previous commit(changes a different file as well) or make a third commit?

rebase onto master and keep it as 2 commits

File: AnkiDroid/src/main/java/com/ichi2/anki/MaterialEditTextDialog.kt [hunk 1]
```diff
@@ -49,7 +50,7 @@ open class MaterialEditTextDialog protected constructor(builder: Builder?) : Mat
          */
         @JvmStatic
         fun displayKeyboard(editText: EditText?, materialDialog: MaterialDialog) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/MaterialEditTextDialog.kt:52]: I feel the issue is that `EditText?` should be `EditText`, does that work?

Comment (codingtosh) [reply AnkiDroid/src/main/java/com/ichi2/anki/MaterialEditTextDialog.kt:52]: So just to confirm, I should make EditText non null only here or everywhere else in the file too?

Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/anki/MaterialEditTextDialog.kt:52]: > So just to confirm, I should make EditText non null only here or everywhere else in the file too?

Just here,

Comment (codingtosh) [reply AnkiDroid/src/main/java/com/ichi2/anki/MaterialEditTextDialog.kt]: Hmm, not sure why it says 4 commits here, despite being ahead only 2 commits master locally. 🤔

File: AnkiDroid/src/main/java/com/ichi2/utils/CollectionUtils.kt [hunk 2]
```diff
@@ -13,25 +13,24 @@
...
-package com.ichi2.utils;
-
-import java.util.Collection;
-import java.util.List;
-
-public class CollectionUtils {
-    /** Throws IndexOutOfBoundsException on empty list*/
-    public static <T> T getLastListElement(List<T> l) {
-        return l.get(l.size()-1);
+object CollectionUtils {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/utils/CollectionUtils.kt:18]: Why was this converted at the same time?

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
Two conversions have been combined here: CollectionUtils and AndroidUiUtils. AndroidUiUtils looks good

**Comment by codingtosh** [comment]
> Two conversions have been combined here: CollectionUtils and AndroidUiUtils. AndroidUiUtils looks good

Ahh, I see what the issue is. I pushed from Android Studio a new branch a little while ago. I think it just pushed the commits to origin master instead of a new branch. I've cleaned it up, PTAL.

**Review by david-allison-1 — APPROVED** [review]
LGTM

**Comment by david-allison-1** [comment]
Did you mean to close this? @codingtosh

**Comment by codingtosh** [comment]
@david-allison-1 No, my bad, I was cleaning up the old ones, didn't realise this one hadn't been merged yet.

**Review by mikehardy — APPROVED** [review]

**Pull request merged**